### ISSUE TITLE
Questions about data generation for REVERIE

### ISSUE THREAD

**Issue by jialuli-luka** [cshizhe/HM3DAutoVLN#1]
Hi shizhe,
    Thanks for the great work!
    I'm trying to generate HM3D data for REVERIE. I've successfully generated the instructions with your provided pre-trained speaker. Could I know what's "pos_vps" in the generated data you provide and would you mind further sharing the code for getting the "pos_vps"? Besides, could I know when sampling the trajectory, is it the case that only one trajectory is sampled for a given object at one viewpoint? 

Jialu

**Comment by cshizhe** [comment]
Hi, the "pos_vps" are the viewpoints where the target object is visible. During HM3D dataset generation, we obtained uniq 3d object id for each environment which can be used to create the "pos_vps". See the released data: HM3D/nav_graphs_v1/features/obj2d_ade20k.
In our work, we only sample one trajectory for a given object at one viewpoint. But you can definetly try to sample more trajectories!

**Comment by jialuli-luka** [comment]
Hi shizhe,
    Thanks for the prompt reply. 
    I'm trying to match the provided object information data in obj2d_ade20k/00000-kfPV7w3FaU5/view_bboxes_merged_by_3d.josnl with the provided instruction-path data in ade20k_pseudo3d_depth2_epoch_94_beam0.jsonl. For scan 00000-kfPV7w3FaU5, it seems that obj0 is detected at viewpoint 0, but in the generated data, instruction id 4_0_0, the "pos_vps" for obj0 only has viewpoint 1 (i.e., viewpoint 0 is not included). I'm not sure I understand it correctly. Would you mind elaborating more on how you get the "pos_vps"? Thank you so much!
Jialu

**Comment by cshizhe** [comment]
Sorry I forgot to mention it in the earlier reply: we select viewpoints where the target object is visible and **close**. We use the distance threshold as 2 meters which achieves the best performance as shown in Table 4 in the paper.

**Comment by jialuli-luka** [comment]
Hi Shizhe, 
    Could I know how you calculate the distance between the objects and viewpoints? Is it the l2 distance between the 3d center of the object and the location of the viewpoint? Besides, Could you please update the view boxes for scan 00011 in your shared dropbox? The folder seems to be empty for 00011. Thank you so much for the help!
Jialu

**Comment by cshizhe** [comment]
I think I used the average depth of the object mask to decide the distance. 
It is weired that the file was not in the folder. I just uploaded it anyway. Thank you for pointing it out.
